apm-sniffer/apm-sdk-plugin/hessian-4.0.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hessian/v4/define/HessianProxyInstrumentation.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.hessian.v4.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+
+import static net.bytebuddy.matcher.ElementMatchers.any;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static org.apache.skywalking.apm.agent.core.plugin.match.NameMatch.byName;
+import static org.apache.skywalking.apm.plugin.hessian.v4.Constants.ADDHEADER_INTERCEPTOR_CLASS;
+import static org.apache.skywalking.apm.plugin.hessian.v4.Constants.CONSTRUCT_INTERCEPTOR;
+import static org.apache.skywalking.apm.plugin.hessian.v4.Constants.ENHANCE_CLASS;
+import static org.apache.skywalking.apm.plugin.hessian.v4.Constants.HESSIAN_SENDREQUEST_CLASS;
+
+/**
+ * enhance method addRequestHeaders
+ *
+ * @author Alan Lau
+ */
+public class HessianProxyInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+
+    @Override protected ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        return new ConstructorInterceptPoint[] {
+
+            new ConstructorInterceptPoint() {
+                @Override public ElementMatcher<MethodDescription> getConstructorMatcher() {
+                    return any();
+                }
+
+                @Override public String getConstructorInterceptor() {
+                    return CONSTRUCT_INTERCEPTOR;
+                }
+            }
+        };
+    }
+
+    @Override protected InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
+        return new InstanceMethodsInterceptPoint[] {
+
+            new InstanceMethodsInterceptPoint() {
+                @Override public ElementMatcher<MethodDescription> getMethodsMatcher() {
+                    return named("addRequestHeaders");
```

Comment:
Enhance this method cannot locate which concrete method is invoked. I think there is arequirement that we prefer to know which concrete method is invoked, but I have no idea how to achieve that yet.

Reply:
Yes, intercept addHeader method is not a good idea, if this method is not being called in all cases.

Reply:
I will check this.

Reply:
I have check,as Hessian client always use HessianProxy to sendRequest. addRequestHeaders always be executed. @wu-sheng @JoshShaw